packages/subproviders/src/subproviders/ledger.ts

```diff
@@ -223,4 +196,39 @@ export class LedgerSubprovider extends BaseWalletSubprovider {
         this._ledgerClientIfExists = undefined;
         this._connectionLock.release();
     }
+    private async _initialDerivedKeyAsync(): Promise<DerivedHDKey> {
+        const ledgerClient = await this._createLedgerClientAsync();
+
+        let ledgerResponse;
+        try {
+            ledgerResponse = await ledgerClient.getAddress(
+                this._derivationBasePath,
+                this._shouldAlwaysAskForConfirmation,
+                SHOULD_GET_CHAIN_CODE,
+            );
+        } finally {
+            await this._destroyLedgerClientAsync();
+        }
+        const hdKey = new HDNode();
+        hdKey.publicKey = new Buffer(ledgerResponse.publicKey, 'hex');
+        hdKey.chainCode = new Buffer(ledgerResponse.chainCode, 'hex');
+        return {
+            hdKey,
+            address: ledgerResponse.address,
+            isChildKey: true,
+            derivationBasePath: this._derivationBasePath,
+            derivationPath: `${this._derivationBasePath}/0`,
+            derivationIndex: 0,
+        };
+    }
+    private _findDerivedKeyByPublicAddress(initalHDKey: DerivedHDKey, address: string): DerivedHDKey {
+        if (_.isUndefined(address) || !addressUtils.isAddress(address)) {
+            throw new Error(WalletSubproviderErrors.FromAddressMissingOrInvalid);
```

Comment:
This error doesn't seem to match up with this helper function... This helper function could be used in many places, and should throw an internal error which is handled by the caller and possibly re-thrown if the `address` fed in was the `from` address of a request. 

Does this make sense?